### PR TITLE
[WIP] Abstract TFormula from ROOT::Double_v

### PR DESCRIPTION
**Pull request by xvallspl**

### CODE REVIEW

File: hist/hist/inc/TFormula.h [hunk 1]
```diff
@@ -177,16 +219,59 @@ class TFormula : public TNamed
...
+         else {
+            if (gDebug)
+               Warning("EvalPar", "Function is not vectorized - converting SIMD types into Double_t and back");
+
+            const int vecSize = vecCore::VectorSize<T>();
+            std::vector<Double_t> xscalars(vecSize * fNdim);
+
+            for (int i = 0; i < vecSize; i++)
+               for (int j = 0; j < fNdim; j++)
+                  xscalars[i * fNdim + j] = vecCore::Get(x[j], i);
```
Comment (amadio) [inline hist/hist/inc/TFormula.h]: Maybe a `vecCore::Gather()` works here? Hardware with AVX2 has gather instructions.

File: hist/hist/inc/TFormula.h [hunk 2]
```diff
@@ -177,16 +219,59 @@ class TFormula : public TNamed
...
+            return DoEval(x, params);
+         }
 #ifdef R__HAS_VECCORE
-   ROOT::Double_v EvalParVec(const ROOT::Double_v *x, const Double_t *params = 0) const;
+         else {
+            if (gDebug)
+               Warning("EvalPar", "Function is not vectorized - converting SIMD types into Double_t and back");
+
+            const int vecSize = vecCore::VectorSize<T>();
+            std::vector<Double_t> xscalars(vecSize * fNdim);
```
Comment (amadio) [inline hist/hist/inc/TFormula.h]: `xscalars` should be aligned to get best performance.

File: hist/hist/inc/TFormula.h [hunk 3]
```diff
@@ -147,9 +149,49 @@ class TFormula : public TNamed
...
-   ROOT::Double_v DoEvalVec(const ROOT::Double_v *x, const Double_t *p = nullptr) const;
-#endif
+
+   ////////////////////////////////////////////////////////////////////////////////
+   /// Evaluate formula.
+   /// If formula is not ready to execute(missing parameters/variables),
+   /// print these which are not known.
+   /// If parameter has default value, and has not been set, appropriate warning is shown.
+
+   template <class T, class NotCompileIfScalarBackend = std::enable_if<!(std::is_same<double, T>::value)>>
```
[the remark this answers is not in the document]
Comment (xvallspl) [reply hist/hist/inc/TFormula.h]: If I'm not mistaken, everything has to be double in the math libraries and functions of ROOT

**Comment by amadio** [comment]
@phsft-bot build

**Comment by bluehood** [comment]
Hi, making a sweep of old PRs -- no activity for almost one year, can this be closed (or merged)?

**Comment by xvallspl** [comment]
Doesn't have an easy solution. Closing it for the moment.

**Pull request closed without merging**